rules/coding-style/src/Rector/Variable/WrapVariableVariableNameInCurlyBracesRector.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\CodingStyle\Rector\Variable;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\PropertyFetch;
+use PhpParser\Node\Expr\Variable;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Core\RectorDefinition\CodeSample;
+use Rector\Core\RectorDefinition\RectorDefinition;
+
+/**
+ * @see \Rector\CodingStyle\Tests\Rector\Variable\WrapVariableVariableNameInCurlyBracesRector\WrapVariableVariableNameInCurlyBracesRectorTest
+ * @see https://www.php.net/manual/en/language.variables.variable.php
+ */
+final class WrapVariableVariableNameInCurlyBracesRector extends AbstractRector
+{
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition('Ensure variable variables are wrapped in curly braces', [
+            new CodeSample(
+                <<<'PHP'
+function run($foo)
+{
+    global $$foo->bar;
+}
+PHP
+                ,
+                <<<'PHP'
+function run($foo)
+{
+    global ${$foo->bar};
+}
+PHP
+            ),
+        ]);
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [Variable::class];
+    }
+
+    /**
+     * @param Variable $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if ($node->name instanceof PropertyFetch) {
+            return new Variable(new PropertyFetch($node->name->var, $node->name->name));
+        } elseif ($node->name instanceof Variable) {
+            return new Variable(new Variable($node->name->name));
+        }
+
```

Comment:
It seems like huge amount of previous code is missing.
What happened?

This will re-create nodes from already correct code. That should not happen.

It's conventions to `return null;` on no-change, so it can report what rule has changed what node.

Reply:
> _It seems like huge amount of previous code is missing.
What happened?_

This is because I tested with both of my examples in the same file and found that only the property fetch was corrected. I had to run Rector again (second pass) before example 2 was corrected. It appears `getStartTokenPos()` gives a different result for `$$value = true;` depending on whether example 1 is `global $$foo->bar;` or `global ${$foo->bar};`

```php
global $$foo->bar; // Example 1

$$value = true; // Example 2
```